### PR TITLE
chore: store freemail domains in pg

### PR DESCRIPTION
**Pull request by Dschoordsch**
# Description

Fixes: #8760 

Store the domains of free email providers in Postgres instead of hardcoding them in the code. Primed the table with the list from Hubspot.
The check `isCompanyDomain` now needs to do a roundtrip to Postgres, so I added a dataloader for this check. This means the dataloader is now passed down quite a lot of helper functions as we check this in many places. Cleanup of these calls is not part of this PR.

## Demo

no demo

## Testing scenarios

- Add some debug output as per comments

- [ ] Create an org with a company domain `@parabol.co`
- [ ] check the active domain is `parabol.co`
- [ ] join with 2 users `@mail.com`, see the active domain did not change
- [ ] join with 2 users `@parabol.fun`, see the active domain did change to `@parabol.fun`

## Final checklist

- [ ] I checked the [code review guidelines](../docs/codeReview.md)
- [ ] I have added [Metrics Representative](../docs/codeReview.md#metrics-representative) as reviewer(s) if my PR invovles metrics/data/analytics related changes
- [ ] I have performed a self-review of my code, the same way I'd do it for any other team member
- [ ] I have tested all cases I listed in the testing scenarios and I haven't found any issues or regressions
- [ ] Whenever I took a non-obvious choice I added a comment explaining why I did it this way
- [ ] I added the label `One Review Required` if the PR introduces only minor changes, does not contain any architectural changes or does not introduce any new patterns and I think one review is sufficient'
- [ ] PR title is human readable and could be used in changelog

### CODE REVIEW

File: packages/server/graphql/mutations/helpers/bootstrapNewUser.ts [hunk 1]
```diff
@@ -30,10 +28,9 @@ const bootstrapNewUser = async (
   const {id: userId, createdAt, preferredName, email, featureFlags, tier, segmentId} = newUser
   // email is checked by the caller
   const domain = email.split('@')[1]!
-  const [isPatient0, usersWithDomain] = await Promise.all([
-    isPatientZero(domain),
-    isCompanyDomain(domain) ? getUsersbyDomain(domain) : []
-  ])
+  const isCompanyDomain = await dataLoader.get('isCompanyDomain').load(domain)
+  const usersWithDomain = isCompanyDomain ? await getUsersbyDomain(domain) : []
+  const isPatient0 = !!domain && isCompanyDomain && usersWithDomain.length === 0
```
Comment (Dschoordsch) [inline packages/server/graphql/mutations/helpers/bootstrapNewUser.ts:33]: I inlined the `isPatientZero` call as this is the only usage and it was doing a second uncached roundtrip to PG to check whether there are other users with the same domain. These users are already loaded in `usersWithDomain`.

Comment (Dschoordsch) [reply packages/server/graphql/mutations/helpers/bootstrapNewUser.ts:33]: Easiest to test this is to add some debug output here

File: packages/server/utils/getActiveDomainForOrgId.ts [hunk 2]
```diff
@@ -1,29 +1,43 @@
...
+  const activeDomain = await pg
+    .selectFrom('User')
+    .leftJoin('FreemailDomain', 'User.domain', 'FreemailDomain.domain')
+    .select(({fn}) => ['User.domain as domain', fn.count('id').as('total')])
+    .where('User.id', 'in', userIds)
+    .where('FreemailDomain.domain', 'is', null)
+    .groupBy('User.domain')
+    .orderBy('total', 'desc')
+    .limit(1)
+    .executeTakeFirst()
```
Comment (Dschoordsch) [inline packages/server/utils/getActiveDomainForOrgId.ts:26]: Again, adding debug output seems to be the easiest way to test this

**Review by Dschoordsch — COMMENTED** [review]
Add comments for testing

**Review by nickoferrall — APPROVED** [review]
Tested and it's working well for me

File: packages/server/utils/getActiveDomainForOrgId.ts [hunk 3]
```diff
@@ -1,29 +1,43 @@
...
+  console.log('GEORG activeDomain', activeDomain)
```
[the remark this answers is not in the document]
Comment (Dschoordsch) [reply packages/server/utils/getActiveDomainForOrgId.ts:27]: Ah, shoot, cleanup ended up in another PR.

**Pull request merged**